### PR TITLE
Docs improvements 2

### PR DESCRIPTION
**Pull request by seeker89**
## TL;DR

Refresh the docs and make it easier to start with the seal.

## Details

- [x] revamp getting started
  - add info about virtualenv
  - show example command to start seal pointing to a kubectl in interactive mode
  - show example command with a basic policy file killing random pods from all namespaces
- [x] add a section on writing policies
   - write up a tutorial on how to write a policy, with some examples
- [x] revamp setup
  - cleanup the arguments which are no longer needed
  - remove unnecessary arguments from the minikube examples 
  - make it easy to understand that you can run with a cloud driver or without
- <del>add some kind of architecture/concepts/terminology section</del>
- [x] integrate docs-schema
- [x] explain when/why you should use each mode
- [x] fix all the relative links 

Resolves https://github.com/bloomberg/powerfulseal/issues/233
Resolves https://github.com/bloomberg/powerfulseal/issues/240

### CODE REVIEW

File: docs/policies/deployment-redundancy.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+              hour: 10
+              minute: 0
+              second: 0
+            endTime:
+              hour: 12
+              minute: 30
+              second: 0
+
+        - randomSample:
+            size: 1
```
Comment (jcstanaway) [inline docs/policies/deployment-redundancy.md:58]: But if one pod was already down for some other reason, this will result in 2 pods being down.

A better policy would be to use:
```
- property:
    name: state
    value: Running
- randomSample:
    ratio: 0.34
```

So if there were only 2 running pods at the time this scenario was executed, the `property` filter brings the list of pods down to 2.  Then the `randomSample.ratio` brings the list of pods down to (0.34 * 2) rounded down to zero.

Doesn't completely eliminate the possibility of "co-chaos", but should help to minimize the possibility.

File: docs/2_getting-started.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+powerfulseal --help
+```
+
+If you have a [`kubeconfig`](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) file ready (for example in `~/.kube/config`) and working with `Kubernetes`, you can start the seal on defaults:
+
+```sh
+powerfulseal interactive
+```
+
+This will start a command line, interactive CLI, just like the following. You can type commands inside of it.
```
Comment (dgzlopes) [inline docs/2_getting-started.md:66]: Maybe a one-liner explaining Interactive mode?

File: docs/2_getting-started.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+2020-06-05 12:53:27 INFO action_nodes_pods.Hello chaos! Matching 'namespace' {'namespace': 'kube-system'}
+2020-06-05 12:53:27 INFO action_nodes_pods.Hello chaos! Matched 37 pods in namespace kube-system
+2020-06-05 12:53:27 INFO action_nodes_pods.Hello chaos! Initial set length: 37
+2020-06-05 12:53:27 INFO action_nodes_pods.Hello chaos! Filtered set length: 37
+2020-06-05 12:53:27 ERROR action_nodes_pods.Hello chaos! Expected pod in state 'Running', got 'ContainerCreating' ([pod #14 name=kube-apiserver-metrics-f98lw namespace=kube-system containers=1 ...])
+2020-06-05 12:53:27 WARNING scenario.Hello chaos! Step returned failure {'podAction': {'matches': [{'namespace': 'kube-system'}], 'actions': [{'checkPodState': {'state': 'Running'}}]}}. Finishing scenario early
+2020-06-05 12:53:27 ERROR policy_runner Exiting early
+2020-06-05 12:53:27 ERROR __main__ Policy runner finishes with an error
+```
+
```
Comment (dgzlopes) [inline docs/2_getting-started.md:191]: I would add here, that autonomous mode is the most used/more configurable mode... Or something like that.

File: docs/5_setup.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+If you're running outside of your cluster, the setup will involve:
+
+- pointing PowerfulSeal at your Kubernetes cluster by giving it a Kubernetes config file
+- pointing PowerfulSeal at your cloud by specifying the cloud driver to use and providing credentials
+- making sure the Seal can `SSH` into the nodes in order to execute `docker kill` command
+- writing a set of policies
+
+It should look something like [this](https://github.com/bloomberg/powerfulseal/blob/master/docs/media/setup.png).
+
```
Comment (dgzlopes) [inline docs/5_setup.md:51]: We should add somewhere in this section a link to the cloud-provider specific set-up docs.

File: docs/7_faq.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+---
+
+## FAQ
+
+### Where can I learn more about Chaos Engineering ?
+
+We found these two links to be a good start:
+
+- [http://principlesofchaos.org/](http://principlesofchaos.org/)
```
Comment (dgzlopes) [inline docs/7_faq.md:26]: http->https

File: docs/policies/deployment-redundancy.md (hunk 1 above)
Comment (seeker89) [reply docs/policies/deployment-redundancy.md:58]: Good point. Added to the doc.

**Comment by dgzlopes** [comment]
I'm taking my time to review the PR. Tons of files :smile: :rocket:

**Comment by seeker89** [comment]
> I'm taking my time to review the PR. Tons of files 😄 🚀

BTW, @dgzlopes I like your idea for the asciicinema - do you think you could help with that bit?

**Comment by dgzlopes** [comment]
> > I'm taking my time to review the PR. Tons of files smile rocket
> 
> BTW, @dgzlopes I like your idea for the Asciicinema - do you think you could help with that bit?

Yeah, sure. I was thinking about recording it locally and uploading it to the provider (Asciinema or another... they are all similar). Then we can embed it in our new docs :)

Also, we can keep the recording versioned in git, so we can make small changes without having to re-record it.

I can work on it tomorrow morning.

**Comment by seeker89** [comment]
> > > I'm taking my time to review the PR. Tons of files smile rocket
> > 
> > 
> > BTW, @dgzlopes I like your idea for the Asciicinema - do you think you could help with that bit?
> 
> Yeah, sure. I was thinking about recording it locally and uploading it to the provider (Asciinema or another... they are all similar). Then we can embed it in our new docs :)
> 
> Also, we can keep the recording versioned in git, so we can make small changes without having to re-record it.
> 
> I can work on it tomorrow morning.

That sounds fantastic 👍

File: docs/7_faq.md (hunk 5 above)
Comment (seeker89) [reply docs/7_faq.md]: Thanks!

File: docs/5_setup.md (hunk 4 above)
Comment (seeker89) [reply docs/5_setup.md:51]: Yup

File: docs/2_getting-started.md (hunk 3 above)
Comment (seeker89) [reply docs/2_getting-started.md:191]: Done

File: docs/2_getting-started.md (hunk 2 above)
Comment (seeker89) [reply docs/2_getting-started.md]: Aye
